clusterloader2/pkg/measurement/common/network/network_performance_measurement.go

```diff
@@ -0,0 +1,449 @@
+	networkPerformanceMetricsName = "NetworkPerformanceMetrics"
+	netperfNamespace              = "netperf"
+)
+
+const (
+	manifestPathPrefix                  = "$GOPATH/src/k8s.io/perf-tests/clusterloader2/pkg/measurement/common/network/manifests"
+	workerPodDeploymentManifestFilePath = manifestPathPrefix + "/" + "*deployment.yaml"
+	networkTestRequestFilePath          = manifestPathPrefix + "/" + "networktestrequests.yaml"
+	crdManifestFilePath                 = manifestPathPrefix + "/" + "*CustomResourceDefinition.yaml"
+	clusterRoleBindingFilePath          = manifestPathPrefix + "/" + "roleBinding.yaml"
+	customResourceDefinitionName        = "networktestrequests.clusterloader.io"
+	rbacName                            = "networktestrequests-rbac"
+)
+
+var (
+	crdGvk = schema.GroupVersionKind{
+		Group:   "apiextensions.k8s.io",
+		Kind:    "CustomResourceDefinition",
+		Version: "v1",
+	}
+
+	rbacGvk = schema.GroupVersionKind{
+		Group:   "rbac.authorization.k8s.io",
+		Kind:    "ClusterRoleBinding",
+		Version: "v1",
+	}
+
+	crGvk = schema.GroupVersionKind{
+		Group:   "clusterloader.io",
+		Kind:    "NetworkTestRequest",
+		Version: "v1alpha1",
+	}
+)
+
+func init() {
+	klog.Info("Registering Network Performance Measurement")
+	if err := measurement.Register(networkPerformanceMetricsName, createNetworkPerformanceMeasurement); err != nil {
+		klog.Fatalf("Cannot register %s: %v", networkPerformanceMetricsName, err)
+	}
+}
+
+func createNetworkPerformanceMeasurement() measurement.Measurement {
+	return &networkPerformanceMeasurement{}
+}
+
+type networkPerformanceMeasurement struct {
+	k8sClient         kubernetes.Interface
+	dynamicClient     dynamic.Interface
+	resourceInterface dynamic.ResourceInterface
+	framework         *framework.Framework
+	numberOfServers   int
+	numberOfClients   int
+	podRatioType      string
+	testDuration      int
+	protocol          string
+	templateMapping   map[string]interface{}
+	// workerPodList stores list of podData for every worker node.
+	podInfo workerPodInfo
+	// metricVal stores MetricResponse received from worker node.
+	metricVal  [][]float64
+	metricLock sync.Mutex
+	stopCh     chan struct{}
+	// startTimeStampForTestExecution is a futureTimeStamp sent to all pods to start
+	// sending the traffic at the same time.
+	startTimeStampForTestExecution int64
+	resultSummary                  testResultSummary
+}
+
+func (npm *networkPerformanceMeasurement) Execute(config *measurement.Config) ([]measurement.Summary, error) {
+	action, err := util.GetString(config.Params, "action")
+	if err != nil {
+		return nil, err
+	}
+	switch action {
+	case "start":
+		if err = npm.validate(config); err != nil {
+			return nil, err
+		}
+		return nil, npm.start(config)
+	case "gather":
+		summary, err := npm.gather()
+		if err != nil {
+			return nil, err
+		}
+		return []measurement.Summary{summary}, err
+	default:
+		return nil, fmt.Errorf("unknown action")
```

Comment:
fmt.Errorf("unknown action: %v", action)

[otherwise it's hard to debug]

Reply:
modified